tests/jballerina-unit-test/src/test/resources/test-src/reachability-analysis/reachability_analysis.bal

```diff
@@ -1176,6 +1176,73 @@ function testReachabilityWithQueryAction() {
     assertEqual(testReachabilityWithQueryAction3(), "Hello");
 }
 
+function fn1(float? v) returns int {
+    int y = 0;
+
+    if v is () {
+        return 1;
+    } else if v is float {
+
+    } else {
+        panic error("err!");
+    }
+
+    return y;
+}
+
+function fn2(float? v) returns int {
+    int y = 0;
+
+    if v is float? {
+
+    } else {
+        panic error("err!");
+    }
+
+    return y;
+}
+
+function fn3(float? v) returns int|error {
+    int y = 0;
+
+    if v is () {
+        fail error("Err");
+    } else if v is float {
+
+    } else {
+        panic error("err!");
+    }
+
+    return y;
+}
+
+function fn4(float? v) returns int {
+    int y = 0;
+
+    if v is () {
+        panic error("Error");
+    } else if v is float {
+
+    } else {
+        panic error("err!");
+    }
+
+    return y;
```

Comment:
I'm not sure if this works atm, but we should ideally have tests like

```ballerina
    int y;

    if v is () {
        y = 1;
    } else if v is float {
        y = 2;
    } else {
        panic error("err!");
    }

    return y;
```

Reply:
This works as expected with this fix. Will add tests and address the above issue via a separate PR.